code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -54,10 +54,13 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
     }
 
     protected void setDns(Instance instance, Map<String, Object> labels, Map<Object, Object> data) {
+        if (instance.getSystemContainer() != null) {
+            return;
+        }
         if (InstanceConstants.KIND_CONTAINER.equals(instance.getKind())) {
             boolean addDns = true;
             if (labels.containsKey(SystemLabels.LABEL_USE_RANCHER_DNS)
-                    && !Boolean.valueOf(SystemLabels.LABEL_USE_RANCHER_DNS)) {
+                    && !Boolean.valueOf(labels.get(SystemLabels.LABEL_USE_RANCHER_DNS).toString())) {
```

Comment:
Also while testing, found this embarrassing bug. Might be one of my late-GA-nights- checkins